File: javascript/gift_certificate/createGiftCertificate.js
import {DateTime} from 'luxon';
import {giftCertificateApi} from '../api.js';

export class CreateGiftCertificate {
    static async execute() {
        const giftCertificate = await CreateGiftCertificate.createGiftCertificateCall();
        console.log("Gift Certificate:", giftCertificate);
    }

    static async createGiftCertificateCall() {

        const giftCertificateCreateRequest = {
            amount: 200.00,
            initial_ledger_description: "Created via TypeScript SDK",
            merchant_note: "Internal comment here",
            email: "support@ultracart.com",
            expiration_dts: DateTime.now()
                .setZone('America/New_York')
                .plus({months: 3})
                .toISO()
        };

        const request = {
            giftCertificateCreateRequest: giftCertificateCreateRequest
        };

        // create does not take an expansion variable.  it will return the entire object by default.
        const gcResponse = await new Promise((resolve, reject) => {
            giftCertificateApi.createGiftCertificate(request, function (error, data, response) {
                if (error) {
                    reject(error);
                } else {
                    resolve(data, response);
                }
            });
        });
        return gcResponse.gift_certificate;
    }
}

export default CreateGiftCertificate;

File: csharp/gift_certificate/CreateGiftCertificate.cs
using System;
using com.ultracart.admin.v2.Api;
using com.ultracart.admin.v2.Model;

namespace SdkSample.gift_certificate
{
    // ReSharper disable once ClassNeverInstantiated.Global
    public class CreateGiftCertificate
    {

        public static void Execute()
        {
            var giftCertificate = CreateGiftCertificateCall();
            Utility.DumpObject(giftCertificate, "Gift Certificate");
        }

        // ReSharper disable once MemberCanBePrivate.Global
        public static GiftCertificate CreateGiftCertificateCall()
        {
            var api = new GiftCertificateApi(Constants.API_KEY);
            
            GiftCertificateCreateRequest createRequest = new GiftCertificateCreateRequest()
            {
                Amount = new Decimal(200.00),
                InitialLedgerDescription = "Created via C# SDK",
                MerchantNote = "Internal comment here",
                Email = "support@ultracart.com",
                ExpirationDts = DateTime.UtcNow.AddMonths(3).ToString("s", System.Globalization.CultureInfo.InvariantCulture)
            };

            // create does not take an expansion variable.  it will return the entire object by default.
            var gcResponse = api.CreateGiftCertificate(createRequest);
            return gcResponse.GiftCertificate;
        }
    }
}